### PR TITLE
plan: refine the handle check to use point get plan for prepare statement

### PR DESCRIPTION
**Pull request by zz-jason**
### What problem does this PR solve? 

for prepare statement, the type fo datum is not int

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: plan/point_get_plan.go [hunk 1]
```diff
@@ -188,15 +188,19 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetP
 			return nil
 		}
 		p := newPointGetPlan(ctx, schema, tbl)
-		p.Handle = handleDatum.GetInt64()
+		var err error
+		p.Handle, err = handleDatum.ToInt64(ctx.GetSessionVars().StmtCtx)
```
Comment (jackysp) [inline plan/point_get_plan.go]: We do not need to convert int64 to int64.

**Comment by coocood** [comment]
@zz-jason 
We need to add a explain test.

File: plan/point_get_plan.go (hunk 1 above)
Comment (zz-jason) [reply plan/point_get_plan.go]: this check is handled in the `ToInt64()` function

File: plan/point_get_plan.go [hunk 2]
```diff
@@ -193,15 +193,19 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetP
...
```
[the remark this answers is not in the document]
Comment (zz-jason) [reply plan/point_get_plan.go]: in prepare statements, the retuned datum is a string, represent the value of the user variable.

File: plan/point_get_plan.go [hunk 3]
```diff
@@ -193,15 +193,19 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetP
...
+		}
 		return p
 	}
 	for _, idxInfo := range tbl.Indices {
 		if !idxInfo.Unique {
 			continue
 		}
 		if idxInfo.State != model.StatePublic {
-			return nil
+			continue
```
Comment (zz-jason) [inline plan/point_get_plan.go]: We should skip the non-public indexes to check all the public indexes, so I changed the `return` instruction to `continue`

**Comment by zz-jason** [comment]
@coocood `explain` an `execute` statement is not allowed. Seems we need a unit test.

**Comment by jackysp** [comment]
How about `explain select * from t where id = '1';` if id is an integer? Will it use point get after this pr?

**Pull request merged**